### PR TITLE
Avoid renaming Search Index tables

### PR DESCRIPTION
**Pull request by crisptrutski**
This also avoids a brittle song and dance with non-authoritative atoms in the background. Now we juggle authoritative atoms all the time 😄

### CODE REVIEW

**Comment by crisptrutski** [comment]
* **#50581** <a href="https://app.graphite.dev/github/pr/metabase/metabase/50581?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#50573** <a href="https://app.graphite.dev/github/pr/metabase/metabase/50573?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/metabase/metabase/50573?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/metabase/search/appdb/index.clj [hunk 1]
```diff
@@ -151,21 +153,26 @@
...
-      (reset! reindexing? false))
-    (drop-table! table-name)
-    (maybe-create-pending! table-name)
-    (when-not testing?
-      (activate-pending!)
-      (reset! initialized? true))
-    true))
+  (drop-table! (pending-table))
+  (maybe-create-pending!)
+  (activate-table!))
```
Comment (piranha) [inline src/metabase/search/appdb/index.clj:158]: it's nice how much simpler it became

**Review by sansolovyov — APPROVED** [review]
One thing that's not obvious: you still need to go through dance of creating the table for the instance to know that search index is available?

**Comment by crisptrutski** [comment]
### Merge activity

* **Nov 29, 2:45 AM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/50573).

**Pull request merged**